### DISCUSSION TITLE
宝塔部署指南

### DISCUSSION THREAD

**Discussion by imsyy** [imsyy/home#104] [Show and tell]
> 要在宝塔上部署，你首先需要有一个配置稍微高一点的服务器，否则可能会在部署时失败

## 部署

- 安装 `Node` 环境，这一步可以使用宝塔上的 `Node.js版本管理器` 应用进行安装
  > 安装最新稳定版

  ![image](https://user-images.githubusercontent.com/42232682/224618056-571933c8-8c84-496f-971c-c555117febcd.png)

- 复制仓库地址

  ![image](https://user-images.githubusercontent.com/42232682/224617735-879d4410-7de8-4a4e-98b7-7014d1ffed8a.png)

- 在站点根目录 `wwwroot` 中打开终端，输入 `git clone 上面复制的仓库地址`

  ![image](https://user-images.githubusercontent.com/42232682/224618965-762f052e-b4af-4b7b-b2ba-adc5e7aaca4d.png)

  > 若出现类似下方的报错，请检查客户端与服务端的 `ssh key` 是否匹配，或者是否在 GitHub 中添加服务器对应的 `ssh key`

  ![image](https://user-images.githubusercontent.com/42232682/224908214-a7b4c397-4d70-4430-90cc-87e2fef18a56.png)

  #### 配置 `ssh key`

  - 在终端输入 `ssh-keygen -t rsa -C "你的 GitHub 邮箱"`，然后一路回车

    ![image](https://user-images.githubusercontent.com/42232682/224973007-fc0d73c7-0dff-427d-9712-29182bcdd3a9.png)

  - 打开 `/root/.ssh/id_rsa.pub` 文件并复制其中的内容

    ![image](https://user-images.githubusercontent.com/42232682/224910475-93d5d4e4-50f2-491a-9cf5-e55e97797495.png)

  - 前往 `https://github.com/settings/keys` 页面，并点击 `New SSH key`，将上述文件内容复制进去

    ![image](https://user-images.githubusercontent.com/42232682/224911147-b15d4c5f-1c48-42f0-895a-6d7279be197b.png)

- `cd` 到 `home` 并安装依赖 `yarn install`
  > 如果未安装 `yarn`，请先执行 `npm install yarn -g`

  ![image](https://user-images.githubusercontent.com/42232682/224619618-b45201e0-4f90-4a82-9892-eaef56554c68.png)
  > 如果安装依赖时出现问题，请将报错截图并在此贴回复

- 执行 `yarn build`
 
  > 出现这一步则代表打包完成，打包后的文件默认在 `dist` 目录中

  ![image](https://user-images.githubusercontent.com/42232682/224622928-6c90b2e9-222d-46a4-9794-d22fd8bc287c.png)

  > 若在此处出现如下图所示的报错，请保证服务器内存充足时重试或者在本地计算机进行打包后再上传至服务器

  ![image](https://user-images.githubusercontent.com/42232682/224620367-aa54fa44-3995-4097-9f22-49ab44eec395.png)

- 构建完成后就可前往 **网站-添加站点** 中填写正确的站点根目录即可访问站点了

  ![image](https://user-images.githubusercontent.com/42232682/224623883-98df489f-4033-4840-920d-5eca8ad6ba6d.png)

## 更新

- 后续更新只需执行 `git pull` 和 `yarn build` 即可实现更新，如果改动了文件，请自行解决冲突后再尝试 `git pull`

**Comment by 176822** [comment]
![image](https://user-images.githubusercontent.com/125774535/224697787-b8dfb1e6-972c-4e7f-a92c-a83b4fb0ea87.jpeg)提示这个

**Reply by imsyy** [reply]
命令是 `git clone git@github.com:imsyy/home.git`

**Reply by imsyy** [reply]
更新了下文档，你看看

**Reply by naihua888** [reply]
作者，安装了SSH key后重新执行部署还是会若出现类似下方的报错，请检查客户端与服务端的 是否匹配，或者是否在 GitHub 中添加服务器对应的ssh keyssh key

**Comment by nljie1103** [comment]
如何安装yarn，我分别尝试了两种
# npm install yarn -g
-bash: npm: command not found

 yum -y install yarn
:Error: Failed to download metadata for repo 'rpmforge': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried

**Reply by imsyy** [reply]
更换国内镜像源再试试

**Comment by nljie1103** [comment]
[root@VM-16-12-centos ~]# cd /home/
[root@VM-16-12-centos home]# yarn build
yarn run v1.22.19
error Couldn't find a package.json file in "/home"
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

这个咋解决啊，作者求解

**Reply by nljie1103** [reply]
服务器内存很充足啊

**Reply by ghost** [reply]
你看看是不是你的目录错了 yarn目录应该是/www/wwwroot/home

**Comment by zquend** [comment]
![a154ff3e53dc470c6e0a213d8913aa4](https://user-images.githubusercontent.com/131616425/236687350-a612931e-5174-4751-9643-f54d2ea9a39d.png)
报错这个

**Reply by ghost** [reply]
1.宝塔安装node.js
2.添加环境变量
vi /etc/profile
3. 在文件最后添加以下配置
export NODE_HOME=/www/server/nodejs/v16.16.0/
export PATH=$PATH:$NODE_HOME/bin
4. 刷新配置
source /etc/profile
5.cd /www/wwwroot
git clone git@github.com:imsyy/home.git
6.如果未安装 yarn，请先执行 npm install yarn -g
cd /www/wwwroot/home 
并安装依赖 yarn install
7.执行 yarn build
8.添加站点 中填写正确的站点根目录即可访问站点了
/www/wwwroot/home/dist

**Reply by zquend** [reply]
谢谢已经安装好了

**Reply by ghost** [reply]
客气 能帮到你我很开心

**Reply by loohalh** [reply]
你这个是服务器宝塔安装吗 我的按照这个步骤安装了还是不行  编译啥的都不会报错  但是就是3000端口没有使用，无法打开

**Comment by haoge0748** [comment]
感谢已经部署好了，补充一下获取key的时候遇到的问题

错误提示：Key is invalid. You must supply a key in OpenSSH public key format
翻译：密钥无效。必须提供OpenSSH公钥格式的密钥
操作步骤
1.生成公钥
ssh-keygen -t rsa -C "GitHub账号的注册邮箱"
2.进入路径
vim ~/.ssh/id_rsa.pub
3.复制公钥到GitHub设置SSH and GPG keys中的SSH keys

错误原因：格式不正确
解决方法：
第2步使用如下命令后复制
cat ~/.ssh/id_rsa.pub

**Comment by linziqi123** [comment]
这样怎么办
![CICZ`GAE NT47_YTCA}{J4H](https://github.com/imsyy/home/assets/136903120/095a9939-8c51-4df4-947f-5ecef0ed429d)

**Reply by imsyy** [reply]
这不是安装成功了吗

**Reply by linziqi123** [reply]
但是他打包不了是为什么
![屏幕截图 2023-06-25 133246](https://github.com/imsyy/home/assets/136903120/08869459-7ba5-41b0-b35c-6c5855080376)

**Reply by shenyunuoyan** [reply]
我也是这样

**Reply by RuSheng920** [reply]
一样

**Comment by WhisToday** [comment]
![image](https://github.com/imsyy/home/assets/130399361/9b40aff4-c4ce-45f0-8320-2b6b4feec20b)

帮忙看看怎么解决

**Reply by nljie1103** [reply]
yarn没安装好吧

**Reply by RuSheng920** [reply]
换稳定版的node.js

**Comment by bosszj** [comment]
太难了，太难了，放弃了

**Comment by RuSheng920** [comment]
![{E4804378-3D66-4158-B54A-B1C0DE237A91}](https://github.com/imsyy/home/assets/121843951/e0425f9e-6dc2-4faa-9f5c-adf55751917d)
这个怎么弄啊，内存是够的

**Reply by sunrixet** [reply]
版本太新了，你用低版本的

**Reply by RuSheng920** [reply]
好的，感谢，请问这个是必须改完再部署，部署了就改不了了，还是部署完也能改？我部署完之后改不了配置

**Reply by imsyy** [reply]
基础信息只能在部署前更改，部署后只能调整个性化设置

**Reply by sunrixet** [reply]
问问作者大大，就是你发的字体防盗链更改的那个文档没看懂

**Reply by xliuwl** [reply]
你这个问题怎么解决的

**Reply by imsyy** [reply]
把 .env.explame 复制并重命名为 .env

**Comment by xliuwl** [comment]
![888888](https://github.com/imsyy/home/assets/170300752/cd8ff7a4-7cb9-4eb2-b54f-c6affca26948)
这个怎么办

**Reply by jfidhdb** [reply]
解决了吗

**Reply by xliuwl** [reply]
已经解决

**Comment by IEmaster** [comment]
为什么其他功能都正常就是没有网页播放器呢

**Comment by ghost** [comment]
![QQ20241110-113650](https://github.com/user-attachments/assets/26074e50-cda8-47b9-8304-a22f44187698)
请问遇到这个报错怎么办

**Reply by IEmaster** [reply]
正确安装依赖了么

**Reply by ghost** [reply]
安装了 执行过了yarn install

**Reply by IEmaster** [reply]
这个报错我之前好像也遇到了 貌似就是依赖不全或者版本不对 你是在什么环境下编译的呢

**Reply by ghost** [reply]
Debian 12 Node v22.11.0

**Reply by IEmaster** [reply]
换成18试试

**Reply by ghost** [reply]
还是不行

**Reply by IEmaster** [reply]
我当时部署的环境是node18.20.3 yarn1.22.22 你可以参考一下

**Reply by ghost** [reply]
还有他们说的什么env.example是什么啊

**Reply by ghost** [reply]
好的改了终于可以了

**Reply by IEmaster** [reply]
更改好后重命名为.env 好好看教程跟着教程一步一步走

**Comment by 1024078948** [comment]
现在的问题有点多，天气和音乐都用不了，我不明白为什么一个算单页要弄的这么麻烦，
